server/app/org/scalaexercises/exercises/controllers/UserProgressController.scala

```diff
@@ -23,18 +23,24 @@ import scalaz.concurrent.Task
 import scala.concurrent.ExecutionContext.Implicits.global
 
 import org.scalaexercises.exercises.services.interpreters.FreeExtensions._
+import freestyle._
+import freestyle.implicits._
 
 class UserProgressController(
-    implicit
-    exerciseOps:     ExerciseOps[ExercisesApp],
-    userOps:         UserOps[ExercisesApp],
-    userProgressOps: UserProgressOps[ExercisesApp],
-    T:               Transactor[Task]
-) extends Controller with JsonFormats with AuthenticationModule with ProdInterpreters {
+    implicit exerciseOps: ExerciseOps[ExercisesApp.Op],
+    userOps: UserOps[ExercisesApp.Op],
+    userProgressOps: UserProgressOps[ExercisesApp.Op],
+    T: Transactor[Task]
+) extends Controller
+    with JsonFormats
+    with AuthenticationModule
+    with ProdInterpreters {
 
   def fetchUserProgressBySection(libraryName: String, sectionName: String) =
     AuthenticatedUser { user ⇒
-      userProgressOps.fetchUserProgressByLibrarySection(user, libraryName, sectionName).runFuture map {
+      userProgressOps
+        .fetchUserProgressByLibrarySection(user, libraryName, sectionName)
+        .runFuture map {
```

Comment:
If we get rid of `runFuture` we can just return free programs and `map` over the program instead of the resulting future
https://github.com/scala-exercises/scala-exercises/pull/622/files#r108734869